tests/ast/ast_test_utils.py

```diff
@@ -320,17 +319,13 @@ def notify_full_ast_validation(query_record: QueryRecord, *args, **kwargs):
     # Ensure sql counter uses the original session and not the validation session.
     tests.integ.utils.sql_counter._active_session = session
 
```

Comment:
can we avoid using globals?

Reply:
I refactored it to be more encapsulated.  The thing is we have a lot of usages of SqlCounter() throughout the test code, and what happens is the full ast validation path will trigger additional queries which then get added to the count, so we need a way to suppress this from happening when executing the validation queries.  Rather than directly interact with globals use TLS state instead which is more friendly for parallel test execution as well.